src/Parser/BufferArray.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @description Stores a collection of buffers as a linked list.
+ */
+class BufferArray {
+  constructor() {
+    this.init();
+
+    this._bufferFunction =
+      typeof module !== "undefined" && module.exports
```

Comment:
What is this checking? Could it be pulled into its own method to make its purpose more clear?

Reply:
This is checking to see if the runtime is a browser vs. NodeJS. Since there is no module in the browser, it should be reasonably safe to use this a runtime environment check. It could probably be moved into the `init()` method, but it's really only intended to run once, and then future calls to the `_newBuffer()` method use the correct buffer function set here.